primitives/core/src/offchain/mod.rs

```diff
@@ -179,6 +179,119 @@ impl TryFrom<u32> for HttpRequestStatus {
 	}
 }
 
+/// Opaque type for offchain timers.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, RuntimeDebug, Encode, Decode, PassByInner)]
+#[cfg_attr(feature = "std", derive(Hash))]
+pub struct TimerId(pub u16);
```

Comment:
I don't feel it's needed a single opaque id type for all `Pollable` kinds should be enough. The low-level API is likely not going to be used anyway, so the users will end up with specific `Futures` that they just `block_on` or `await`.